src/sec.c

```diff
@@ -23,11 +23,11 @@ sec_bin_init(sec_bin_t *bin) {
 bool
 sec_init(tsdn_t *tsdn, sec_t *sec, base_t *base, pai_t *fallback,
     const sec_opts_t *opts) {
-	size_t max_alloc = opts->max_alloc & PAGE_MASK;
-	pszind_t npsizes = sz_psz2ind(max_alloc);
-	if (sz_pind2sz(npsizes) > opts->max_alloc) {
-		npsizes--;
-	}
+	assert(opts->max_alloc > 0);
+
+	size_t max_alloc = opts->max_alloc & ~PAGE_MASK;
+	pszind_t npsizes = sz_psz2ind(max_alloc) + 1;
```

Comment:
OK, maybe I'm about to propose the exact same bug I already introduced the first time around, but: shouldn't we have
```
if (sz_pind2psz(sz_psz2ind(max_alloc)) > max_alloc) {
  npsizes--;
}
```
after this?

Otherwise, if  max_alloc is between two size classes, I think we'll allow sz_psz2ind(max_alloc)-sized extents to be cached, violating the max guarantee.

Reply:
No, because we decide whether to cache an extent based on `size <= sec->opts.max_alloc`, not `npsizes`. `npsizes` only describes how many bins we have. If `max_alloc` is between bin 9 and bin 10 (zero based), we want to have 11 bins, because otherwise we won't have bin 10, but may need to cache an extent that falls in that bin.